maestro-client/src/main/java/maestro/Filters.kt

```diff
@@ -54,26 +54,41 @@ object Filters {
         nodes.filter { this(it) }
     }
 
-    fun textMatches(text: String): ElementLookupPredicate {
-        return {
-            it.attributes["text"]?.let { value ->
-                text == value
-                    || text == value.replace('\n', ' ')
-            } ?: false
-        }
+    fun textMatches(regex: Regex): ElementLookupPredicate {
```

Comment:
We need to match in order of priority:

- Text
- Hint
- Accessibility

Otherwise, this risks being a breaking change. 

For inspiration how to do it, take a look at `idMatches` method that returns `ElementFilter` instead of `ElementLookupPredicate`

Reply:
Why would this be a breaking change. If someone uses hint already it should match same for content description.

Am I missing something?

Reply:
Say you have 3 views in your hierarchy that appear in this order:

- `View(hint="Hello World")`
- `View(accessibilityText="Hello World")`
- `View(text="Hello World")`

In previous implementation a third view will be matched. In your implementation a first view will be matched. This is a breaking change. First or second view should only match if there is no third view.

Reply:
Done ✅